pyogrio/_geometry.pyx

```diff
@@ -101,18 +101,24 @@ cdef str get_geometry_type(void *ogr_layer):
     return GEOMETRY_TYPES[ogr_type]
 
 
-cdef int get_geometry_type_code(str geometry_type):
+cdef int get_geometry_type_code(str geometry_type, bint promote_to_multitype):
     """Get geometry type code for string geometry type.
 
     Parameters
     ----------
-    geometry_type : str
+    geometry_type: str
+    promote_to_multitype: bint (bool)
 
     Returns
     -------
     int
         geometry type code
     """
+    if promote_to_multitype is True and "Multi" not in geometry_type:
+        geometry_type = (geometry_type
```

Comment:
Suggestion: convert to conditional:

```Python
if geometry_type == 'Point':
    geometry_type = 'MultiPoint'
elif ...
```

I appreciate that the `replace` is a bit more terse, but it doesn't feel like the right operation here.

Reply:
The problem is a bit that it leads to really a lot of if's, as there are many variants of all types: measureds, 3D,...

Do you still prefer conditionals being aware of that?
```
GEOMETRY_TYPES = {
    wkbUnknown: 'Unknown',
    wkbPoint: 'Point',
    wkbLineString: 'LineString',
    wkbPolygon: 'Polygon',
    wkbMultiPoint: 'MultiPoint',
    wkbMultiLineString: 'MultiLineString',
    wkbMultiPolygon: 'MultiPolygon',
    wkbGeometryCollection: 'GeometryCollection',
    wkbNone: None,
    wkbLinearRing: 'LinearRing',
    # WARNING: Measured types are not supported in GEOS and downstream uses
    # these are stripped automatically to their corresponding 2D / 3D types
    wkbPointM: 'PointM',
    wkbLineStringM: 'Measured LineString',
    wkbPolygonM: 'Measured Polygon',
    wkbMultiPointM: 'Measured MultiPoint',
    wkbMultiLineStringM: 'Measured MultiLineString',
    wkbMultiPolygonM: 'Measured MultiPolygon',
    wkbGeometryCollectionM: 'Measured GeometryCollection',
    wkbPointZM: 'Measured 3D Point',
    wkbLineStringZM: 'Measured 3D LineString',
    wkbPolygonZM: 'Measured 3D Polygon',
    wkbMultiPointZM: 'Measured 3D MultiPoint',
    wkbMultiLineStringZM: 'Measured 3D MultiLineString',
    wkbMultiPolygonZM: 'Measured 3D MultiPolygon',
    wkbGeometryCollectionZM: 'Measured 3D GeometryCollection',
    wkbPoint25D: '2.5D Point',
    wkbLineString25D: '2.5D LineString',
    wkbPolygon25D: '2.5D Polygon',
    wkbMultiPoint25D: '2.5D MultiPoint',
    wkbMultiLineString25D: '2.5D MultiLineString',
    wkbMultiPolygon25D: '2.5D MultiPolygon',
    wkbGeometryCollection25D: '2.5D GeometryCollection',
}
```